src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -490,7 +556,23 @@ def table(self, name: Union[str, Iterable[str]], _emit_ast: bool = True) -> Tabl
             ast.reader.CopyFrom(self._ast)
             build_table_name(ast.name, name)
 
-        table = self._session.table(name, _emit_ast=False)
+        if time_travel_mode is not None:
+            direct_param_values = [
+                time_travel_mode,
+                statement,
+                offset,
+                timestamp,
+                timezone,
+                stream,
+            ]
+            time_travel_params = dict(
+                zip(TIME_TRAVEL_OPTIONS_PARAMS_MAP.values(), direct_param_values)
+            )
+            time_travel_params = {k: v for k, v in time_travel_params.items()}
```

Comment:
I don't think you need this line. it's basically deep copying a time_travel_params
```suggestion
```